pymeasure/instruments/keysight/keysightN5767A.py

```diff
@@ -0,0 +1,111 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2017 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+
+import logging
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import (
+    truncated_range, truncated_discrete_set,
+    strict_discrete_set
+)
+
+from pymeasure.adapters import VISAAdapter
+
+class KeysightN5767A(Instrument):
+    """ Represents the Keysight N5767A Power supply
+    interface for interacting with the instrument.
+
+    .. code-block:: python
+
+    """
+    ###############
+    # Current (A) #
+    ###############
+    current_range = Instrument.control(
+        ":CURR?", ":CURR %g",
+        """ A floating point property that controls the DC current range in
+        Amps, which can take values from 0 to 25 A.
+        Auto-range is disabled when this property is set. """,
+        validator=truncated_range,
+        values=[0, 25],
+    )
+
+    current = Instrument.measurement(":MEAS:CURR?",
+        """ Reads a setting current in Amps. """
+    )
+
+    ###############
+    # Voltage (V) #
+    ###############
+    voltage_range = Instrument.control(
+        ":VOLT?", ":VOLT %g V",
+        """ A floating point property that controls the DC voltage range in
+        Volts, which can take values from 0 to 60 V.
+        Auto-range is disabled when this property is set. """,
+        validator=truncated_range,
+        values=[0, 60]
+    )
+
+    voltage = Instrument.measurement("MEAS:VOLT?",
+        """ Reads a DC voltage measurement in Volts. """
+     )
+
+    ###############
+    # State (OFF/ON) #
+    ###############
+    state = Instrument.control(
```

Comment:
I would suggest providing explicit methods for `enable`, `disable`, and `is_enabled` in the same way that it is done in the [Danfysik8500 power supply](https://github.com/ralph-group/pymeasure/blob/master/pymeasure/instruments/danfysik/danfysik8500.py#L99-L112). The strings `ON` and `OFF` are understandable (especially given their direct use in the command), but they seem arbitrary relative to other instrument classes in PyMeasure.

Reply:
I have added enable,disable,is_enabled, and add _status measurement to read power supply status.
As you said ,so users may not care about string "ON"/"OFF“ or 0,1, returns type are boolean.